src/main/java/Byte.java

```diff
@@ -0,0 +1,116 @@
+import java.util.Scanner;
+
+public class Byte {
+    private static final int MAX_TASKS = 100;
+    private static final Task[] tasks = new Task[MAX_TASKS];
+    private static int taskCount = 0;
+
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        runByte(scanner);
+        scanner.close();
+    }
+
+    public static void runByte(Scanner scanner){
+        printWelcomeMessage();
+        while(true){
+            String userInput = scanner.nextLine();
+            printLineSeparator();
+            if (processUserInput(userInput)) {
+                break;
+            }
+        }
+    }
+
+    public static boolean processUserInput(String userInput){
+        if (userInput.equals("bye")) {
+            printGoodbyeMessage();
+            return true;
+        } else if (userInput.startsWith("mark ")) {
+            markTask(userInput.substring("mark ".length()), true);
+        } else if (userInput.startsWith("unmark ")) {
+            markTask(userInput.substring("unmark ".length()), false);
+        } else if (userInput.equals("list")) {
+            listTasks();
+        } else if (userInput.startsWith("todo ")){
+            handleToDoCommand(userInput);
+        }else if(userInput.startsWith("deadline ")){
+            handleDeadlineCommand(userInput);
+        }else if(userInput.startsWith("event ")){
+            handleEventCommand(userInput);
+        }
+        return false;
+    }
+
+    private static void handleToDoCommand(String userInput) {
```

Comment:
Is there a reason why you didn't put handleToDoCommand function under class ToDo, I noticed this for handleDeadline etc

Reply:
Task classes remain dedicated to modelling tasks, not parsing user input or handling application commands which is one of the reasons why I didn't put it under the class but rather on the main method.